### ISSUE TITLE
Support for writing business logic in GoLang or Java

### ISSUE THREAD

**Issue by kagile** [hyperledger/composer#4306]
Hyperledger Fabric can be written in Java and Golang. But, why are they not supported in Hyperledger Composer? Many of us have big issues with Javascript and dynamic typing languages in general. Will Java or Go be supported on Hyperledger Composer some day? Are there any progress on it? If not, can you provide guidelines to contribute for supporting a new language on Hyperledger Composer?

**Comment by akaila** [comment]
How about TypeScript? I am guessing its not easy to do GoLang or Java given the packaging of composer in archive etc. JavaScript is quite simple and TypeScript can provide compile time type safety.

**Comment by davidkel** [comment]
@kagile There are no plans to support business networks written in Go or Java. Client applications can be written in any language utilising the Rest Server.

As for what it would take to support one of the other languages available in Fabric (for which currently Go is the only supported language with Java hopefully coming in future releases) Would require a fair amount of work, requiring a new implementation of composer-runtime and composer-runtime-hlfv1. Also you would need to consider client side access. For example the CLIs are all node modules would you be looking at providing equivalents that don't require node/npm to be installed.

**Comment by kagile** [comment]
@akaila, @davidkel  TypeScript is okay but I don't see any option in the composer. Though I would say TypeScript IS a Javascript.  The reason many don't want to invest in the javascript is the bad design of the language itself along with a fragile community. For the blockchain stuff, most business logic will be critical and everyone wants it to be with the minimum amount of bugs. I wonder why the composer has chosen the path of Javascript!

**Comment by akaila** [comment]
TypeScript is no more or less buggy than Go or Java. However it is more dynamic and I am not going to prove here how that is taking over the world. Composer is a dynamic layer over fabric and statically typed languages are not easy to mould it to that effect. If choice of language trumps your decision, take a look at NEO which supports C#.

**Comment by kagile** [comment]
@akaila @davidkel @nklincoln But at least it provides type safety. At least, you could support TypeScript along with Javascript. There should not be much effort in supporting TypeScript.

**Comment by akaila** [comment]
@kagile I agree with you there. In my projects I had to do quite a bit of fiddling even though you can generate TypeScript models via "composer generate create --format Typescript" it is of little use. Basically you can use it as a reference or as a metadata/type definition (.d.ts) in your code. The biggest blocker is that currently composer does not allow for requiring external files. The way around is to use webpack preserving the method comments.

**Comment by kagile** [comment]
Interesting. However, I don't agree with the statement: "Today it takes ~50 lines of code to submit a transaction using the Fabric APIs, whilst in Composer it takes ~5 lines of code, and that's wrong - Composer's value should not come from just making Fabric easier to use." 
Making something easy is hard. Composer was good at making things easier to understand and build for most people.

**Comment by akaila** [comment]
Wow that drops the bomb on composer and craps it out completely. This is not a roadmap, its a funeral procession to say the least. I was disappointed with the lack of production support from IBM but this short term thinking and canning of promising products takes it at a new low. Good thing I am not using any of the hyperledger products anymore.
